app/models/email_subscription.rb

```diff
@@ -4,11 +4,42 @@ class EmailSubscription < ApplicationRecord
   validates :name, presence: true
   validates :topic_slug, presence: true
 
+  before_destroy :deactivate!
+
   def to_hash
     {
       "name" => name,
       "topic_slug" => topic_slug,
       "email_alert_api_subscription_id" => email_alert_api_subscription_id,
     }.compact
   end
+
+  def reactivate_if_confirmed!(email, email_verified)
+    deactivate!
+
+    return unless email_verified
+
+    subscriber_list = GdsApi.email_alert_api.get_subscriber_list(
+      slug: topic_slug,
+    )
+
+    subscription = GdsApi.email_alert_api.subscribe(
+      subscriber_list_id: subscriber_list.to_hash.dig("subscriber_list", "id"),
+      address: email,
+      frequency: "daily",
+      skip_confirmation_email: true,
+    )
+
+    update!(email_alert_api_subscription_id: subscription.to_hash.dig("subscription", "id"))
+  end
+
+  def deactivate!
+    return unless email_alert_api_subscription_id
+
+    GdsApi.email_alert_api.unsubscribe(email_alert_api_subscription_id)
+    update!(email_alert_api_subscription_id: nil)
+  rescue GdsApi::HTTPGone, GdsApi::HTTPNotFound
```

Comment:
So in this flow,
- We want to deactivate a user before destroying an account
- If the call to `GdsApi.email_alert_api.unsubscribe` returns either `gone` or `not found` then this is rescued and the destroy continues anyway?

Are there any other status codes we would wish to check for?
Is it any better if we wrapped like 40 in something like?

```
if response.code == 200
   update!(email_alert_api_subscription_id: nil)
end
```

Not sure what other possible return values from the API would be, and when it's OK to delete a subscription if we've not sucessfulyl unsubscribed a user?

Reply:
It returns a 204 if the subscription has been ended, and a 404 or 410 if subscription has been ended through some other mechanism (like the user cancelling it themselves through email-alert-frontend).

So actually this code is slightly wrong, it should consider a 404 and a 410 success too - I'll fix the `update!` and add a test.

Reply:
f80df855ae3517220fe616545d166789795e1a6f